### PR TITLE
Fixed event-owner property of email not getting the right AWS username

### PR DESCRIPTION
**Pull request by mechastorm**
This is my first PR - there are several other changes I would like to propose based on this but I am doing one minor change that should be backwards compatible. 

Issue: #1184 

When using `c7n_mailer` to enable email notifications on a policy, I want an email sent directly to the user who created/triggered that event.

Example use case:
I have a policy discouraging doing cidr 0.0.0.0/0 as a security group rule in non-productions. 
I want the policy to email the user who did the change and cc to the Ops teams.
```
policies:
  - name: open-network-sg-rule
    resource: security-group
    mode:
      type: cloudtrail
      role: arn:aws:iam::123456789:role/custodian-monitor
      events:
        - event: AuthorizeSecurityGroupIngress
          source: ec2.amazonaws.com
          ids: requestParameters.groupId
    filters:
      - type: ingress
        Cidr:
          value: 0.0.0.0/0
          op: eq
          value_type: cidr
    actions:
      - type: notify
        template: default.html
        priority_header: 2
        subject: "[Cloud Custodian] Security Group Rule Violation! - Too Open!"
        violation_desc: |
          IP Range to the open world (0.0.0.0/0) should not be used!
          You should only whitelist by Security Group ID or specific single IP
        to:
          - teamemail@org.com
          - event-owner
        transport:
          type: sqs
          queue: https://sqs. us-west-2.amazonaws.com/123456789/email_queue
          region: us-west-2
```

Currently when using the `event-owner` in the `to:` field, the lambda function gets the `PrincipalID` from the `UserIdentity` section of the cloudtrail event. This normally results in the actual IAM Access Key which in turn results in an invalid email sent.

I have now modified this so that it will now instead use the proper IAM username when `event-owner` is used. This still relies on that the IAM username is a valid email address and future changes can adjust for that. 

### Changed
- `tools/c7n_mailer/c7n_mailer/sqs_message_processor.py`
  - Added a new parameter to `get_aws_username_from_event` called `user_identity_field` with a default value of `principalId`. 
    - This parameter is used to denote the property of the cloudtrail event to get the username.
    -  The default value of `principalId` is used to ensure backwards compatibility to other methods that use this.
  - In `process_sqs_messsage` method, the `event-owner` target will now use the correct IAM username.

### CODE REVIEW

File: tools/c7n_mailer/c7n_mailer/sqs_message_processor.py [hunk 1]
```diff
@@ -106,7 +106,7 @@ def process_sqs_messsage(self, message):
             self.send_resource_owner_messages(data)
         if 'event-owner' in targets:
             targets.remove('event-owner')
-            recipient = self.get_aws_username_from_event(data['event'])
+            recipient = self.get_aws_username_from_event(data['event'], "userName")
```
Comment (kapilt) [inline tools/c7n_mailer/c7n_mailer/sqs_message_processor.py]: we probably need to make this configurable, the exact field tends to be dependent on org specific auth setups it seems.

**Comment by kapilt** [comment]
for your auth setup does splitting on the arn work? ie if we rsplit on '/' and grab the last element?

**Comment by mechastorm** [comment]
hi @kapilt, actually the the last path of the userIdentity arn would work for my auth setup. Would that method work in other auth methods? My org doesnt yet use SSO for their AWS so I am not sure how it works for the others. 

If not I can take your initial suggestion and that fieldname configurable as well.

**Comment by kapilt** [comment]
useridentity arn split on last path seems to work for our sso setup as well, if your game let's do that instead.

**Comment by kapilt** [comment]
it looks like this has been subsequently addressed, as getting the aws user from the event has since been greatly expanded in what it supports.

**Pull request closed without merging**